typing/includecore.ml

```diff
@@ -43,6 +43,7 @@ type value_mismatch =
   | Type of Errortrace.moregen_error
   | Zero_alloc of Zero_alloc.error
   | Modality of Mode.Modality.Value.error
```

Comment:
Is this `Modality` case still reachable?

Reply:
No it's no longer reachable. But I think as we introduce non-legacy modules, comparing two module types will be done without knowing modes, so this will be reachable again.